Sources/MapboxDirections/RouteLeg.swift

```diff
@@ -283,6 +273,25 @@ extension RouteLeg: CustomQuickLookConvertible {
     }
 }
 
+extension RouteLeg: NSCopying {
+    public func copy(with zone: NSZone? = nil) -> Any {
```

Comment:
Since #382, we’ve avoided implementing NSCopying, because in the past, we’ve frequently forgotten to add to copy methods when adding new properties, resulting in silent data loss. The hope expressed in #381 was that we’d convert the classes to structs, which have built-in copying functionality. In the meantime, the navigation SDK worked around the lack of NSCopying conformance in RouteOptions by round-tripping through JSON:

https://github.com/mapbox/mapbox-navigation-ios/blob/5f0642fa1574450302baa7fa0f8683df39e5bfdd/MapboxCoreNavigation/RouteOptions.swift#L31-L35

But I can see how that approach would be wildly unperformant for copying an entire leg. Do we really need to copy each leg and create a new route when refreshing it? By making the segment* properties mutable, we’ve already set the expectation that they can be changed out from under client code. I think it would be simpler to just set those properties on the original object.

Reply:
> Do we really need to copy each leg and create a new route when refreshing it?

Looks like we previously discussed this point in https://github.com/mapbox/mapbox-directions-swift/pull/420#discussion_r433385717.